PSModule/AbstractLauncherTaskCmdlet.cs

```diff
@@ -451,6 +410,47 @@ protected virtual bool CollateResults(string resultFile, string log, string resd
             return true;
         }
 
+        private bool HasResults(string resultFile, out string xmlResults)
+        {
+            xmlResults = null;
+            if (!File.Exists(resultFile))
+            {
+                WriteDebug("result file not found");
+                return false;
+            }
+
+            //read result xml file
+            xmlResults = File.ReadAllText(resultFile);
+
+            if (xmlResults.IsNullOrWhiteSpace())
+            {
+                WriteDebug("Empty results file");
+                return false;
+            }
+            else
+            {
+                try
+                {
+                    var doc = XDocument.Parse(xmlResults);
+                    if (doc?.Root == null || !doc.Root.HasElements)

```

Comment:
Shall we also check the **root** element? For example, the name of the root element.

Reply:
At least at this step, we only need to know if any xml results exist.
Eventually when the xml content will be interpreted we can check it deeper.
Currently we didn't find a test scenario where HpToolsLauncher would return an xml with invalid root name.